app/models/manageiq/providers/azure/cloud_manager/metrics_capture.rb

```diff
@@ -1,165 +1,176 @@
-class ManageIQ::Providers::Azure::CloudManager::MetricsCapture < ManageIQ::Providers::BaseManager::MetricsCapture
-  INTERVAL_1_MINUTE = "PT1M".freeze
-
-  # Linux, Windows counters
-  CPU_METERS     = ["\\Processor Information(_Total)\\% Processor Time",
-                    "\\Processor(_Total)\\% Processor Time",
-                    "\\Processor\\PercentProcessorTime"].freeze
+# frozen_string_literal: true
 
-  NETWORK_METERS = ["\\NetworkInterface\\BytesTotal"].freeze # Linux (No windows counter available)
+## https://docs.microsoft.com/en-us/azure/azure-monitor/platform/metrics-supported
+# ..#microsoftclassiccomputevirtualmachines
+# ..#microsoftcomputevirtualmachines
 
-  MEMORY_METERS  = ["\\Memory\\PercentUsedMemory",
-                    "\\Memory\\% Committed Bytes In Use",
-                    "\\Memory\\Committed Bytes"].freeze
-
-  DISK_METERS    = ["\\PhysicalDisk\\BytesPerSecond",
-                    "\\PhysicalDisk(_Total)\\Disk Read Bytes/sec", # Windows
-                    "\\PhysicalDisk(_Total)\\Disk Write Bytes/sec",
-                    "\\LogicalDisk(_Total)\\Disk Read Bytes/sec",
-                    "\\LogicalDisk(_Total)\\Disk Write Bytes/sec"].freeze # Windows
+class ManageIQ::Providers::Azure::CloudManager::MetricsCapture < ManageIQ::Providers::BaseManager::MetricsCapture
+  INTERVAL_1_MINUTE = 'PT1M'
+  private_constant :INTERVAL_1_MINUTE
 
-  COUNTER_INFO = [
+  COUNTERS_INFO = [
+    ## CPU
     {
-      :native_counters       => CPU_METERS,
-      :calculation           => ->(stat) { stat.first },
-      :vim_style_counter_key => "cpu_usage_rate_average"
+      :counter_key => 'cpu_usage_rate_average',
+      :source      => :api,
+      :metrics     => [
+        ['Percentage CPU'].freeze,
+      ].freeze,
+      :calculation => ->(stats) { stats.mean },
+      :unit_key    => 'percent',
     },
+    ## Memory
     {
-      :native_counters       => NETWORK_METERS,
-      :calculation           => ->(stat) { stat.first / 1.kilobyte },
-      :vim_style_counter_key => "net_usage_rate_average",
+      :counter_key => 'mem_usage_rate_average',
+      :source      => :raw,
+      :metrics     => [
+        ['/builtin/memory/percentusedmemory'].freeze, # linux
+        ['\Memory\% Committed Bytes In Use'].freeze, # windows
+      ].freeze,
+      :calculation => ->(stats) { stats.mean },
+      :unit_key    => 'percent',
     },
+    ## Disk
     {
-      :native_counters       => MEMORY_METERS,
-      :calculation           => ->(stat) { stat.first },
-      :vim_style_counter_key => "mem_usage_absolute_average",
+      :counter_key => 'disk_usage_absolute_average',
+      :source      => :api,
+      :metrics     => [
+        ['Per Disk Read Bytes/sec', 'Per Disk Write Bytes/sec'].freeze, # Microsoft.Compute/virtualMachines
+        ['Disk Read Bytes/Sec', 'Disk Write Bytes/Sec'].freeze, # Microsoft.ClassicCompute/virtualMachines
```

Comment:
We should not care about classicCompute at all.

Reply:
great! :)